### PR TITLE
Improve check_tidy script

### PR DESCRIPTION
**Pull request by galpeter**
List of improvements:
* Now works with both 3.x and 2.7+ Python versions.
* Separate classes for general style and clang-format checks.
* Allowed CMakeLists.txt for style checks.
* The diff for clang-format uses temp files.
* The diff for clang-format is now a unified diff.
* Uses clang-format if clang-format-3.8 is not found.
* If no clang-format found still run the other checks.
* Executor.run_cmd_output now accepts an list instead of a string.

IoT.js-DCO-1.0-Signed-off-by: Peter Gal pgal.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by yichoi** [comment]
LGTM

File: tools/check_tidy.py [hunk 1]
```diff
@@ -14,113 +14,166 @@
...
-            if is_checked_by_clang(file):
-                formatted = ex.run_cmd_output(' '.join(['clang-format-3.8',
-                                                        '-style=file', file]),
-                                              quiet=True)
-                f = open(file + '.formatted', 'w')
-                f.write(formatted)
-                f.close()
-                ex.check_run_cmd('diff', [file, file+'.formatted'], quiet=True)
-                fs.remove(file + '.formatted')
+    exts_allowed = ['.c', '.h', '.js', '.py', '.sh', '.cmake']
```
Comment (LaszloLango) [inline tools/check_tidy.py]: What about .txt files (e.g: CMakeLists.txt)?

**Comment by yichoi** [comment]
sounds reasonable.

File: tools/check_tidy.py (hunk 1 above)
Comment (galpeter) [inline tools/check_tidy.py]: That's a good question. I'm using the same extensions which were already in place. Maybe someone more knowledgeable could answer that question.

**Comment by galpeter** [comment]
Question about remove or update the merge.py was my next item :). So should I remove the merge.py changes from this file and create another PR which removes the merge.py file?

**Comment by LaszloLango** [comment]
@galpeter, yes. It is unnecessary to change anything in 'merge.py', because it is obsolete. Just remove it in your next PR.

File: tools/check_tidy.py (hunk 1 above)
Comment (LaszloLango) [inline tools/check_tidy.py]: We should check CMakeLists.txt files if we check *.cmake files too. But allowing .txt files might be too general and could conflict with other things, likes test resources.

**Comment by galpeter** [comment]
@LaszloLango PR is updated.
Changes:
* CMakeLists.txt now added for general style checking.
* Removed changes in the merge.py.

**Comment by LaszloLango** [comment]
@galpeter, thanks for the update. LGTM. BTW the changes in merge.py are still there. :) But it does not matter, because we'll remove it anyway.

@yichoi, please check the changes since your last review.

**Comment by galpeter** [comment]
@LaszloLango strange... I removed them. Just a sec I'll fix it.

**Comment by galpeter** [comment]
@LaszloLango done. Should be ok now,

**Comment by yichoi** [comment]
still LGTM

**Pull request merged**